tests/test_utils.py

```diff
@@ -965,6 +967,26 @@ def get_connection_string_and_dialect(
     return dialect, connection_string
 
 
+def add_datasource(
```

Comment:
Attention reviewer! I don't like this, but I think it's the best we can do without restructuring tests more or passing method names as strings 🤮 . Open to suggestions if there's something better we can do!